src/app/shared/search-box/input/suggestions/search-suggestions.component.ts

```diff
@@ -0,0 +1,171 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges} from '@angular/core';
+import {Store} from '@ngrx/store';
+import {Observable} from 'rxjs/Observable';
+import {catchError, debounceTime, map, startWith, switchMap, withLatestFrom} from 'rxjs/operators';
+import {Subject} from 'rxjs/Subject';
+import {Subscription} from 'rxjs/Subscription';
+import {Suggestions, SuggestionType} from '../../../../core/dto';
+import {SearchService} from '../../../../core/rest';
+import {AppState} from '../../../../core/store/app.state';
+import {selectAllCollections} from '../../../../core/store/collections/collections.state';
+import {FulltextQueryItem} from '../../query-item/model/fulltext.query-item';
+import {QueryItem} from '../../query-item/model/query-item';
+import {QueryItemType} from '../../query-item/model/query-item-type';
+import {SuggestionsConverter} from './suggestions.converter';
+
+@Component({
+  selector: 'search-suggestions',
+  templateUrl: './search-suggestions.component.html',
+  styleUrls: ['./search-suggestions.component.scss'],
+  // changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SearchSuggestionsComponent implements OnChanges, OnDestroy, OnInit {
+
+  @Input()
+  public queryItems: QueryItem[] = [];
+
+  @Input()
+  public text: string;
+
+  @Input()
+  public moveSelection$: Observable<number>;
+
+  @Input()
+  public useSelection$: Observable<string>;
+
+  @Output()
+  public useSuggestion = new EventEmitter<QueryItem>();
+
+  public suggestions: QueryItem[] = [];
+  public selectedIndex = -1;
+
+  public suggesting = true;
+
+  private searchTerms$ = new Subject<string>();
+  private suggestionsSubscription: Subscription;
+
+  private moveSelectionSubscription: Subscription;
+  private useSelectionSubscription: Subscription;
+
+  constructor(private searchService: SearchService,
+              private store: Store<AppState>) {
+  }
+
+  public ngOnInit() {
+    this.suggestQueryItems();
+    this.subscribeToMoveSelection();
+    this.subscribeToUseSelection();
+  }
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.hasOwnProperty('queryItems') && this.queryItems) {
+      this.suggesting = !this.queryItems.find(queryItem => queryItem.type === QueryItemType.View);
+    }
+    if (changes.hasOwnProperty('text')) {
+      if (this.text) {
+        this.searchTerms$.next(this.text);
+      } else {
+        this.updateSuggestions([]);
+      }
+    }
+  }
+
+  public ngOnDestroy() {
+    if (this.moveSelectionSubscription) {
+      this.moveSelectionSubscription.unsubscribe();
+    }
+    if (this.useSelectionSubscription) {
+      this.useSelectionSubscription.unsubscribe();
+    }
+    if (this.suggestionsSubscription) {
+      this.suggestionsSubscription.unsubscribe();
+    }
+  }
```

Comment:
Look at documentation, you can use one subscription for all observables:
https://github.com/ReactiveX/rxjs/blob/master/doc/subscription.md

Reply:
Interesting. But I'm not sure if we should use it. Look at [this discussion](https://github.com/ReactiveX/rxjs/issues/2769). I don't really know which way we can use it safely.